filebeat/input/filestream/fswatch.go

```diff
@@ -168,6 +173,16 @@ func (w *fileWatcher) watch(ctx unison.Canceler) {
 			writtenCount++
 		}
 
+		// For the delete feature we need to run the harvester for
+		// files that have not changed until they're deleted.
+		//
+		// Op.Done is the default value for a loginp.Operation,
+		// it is used below to indicate the file has not changed and
+		// by the copytruncate prospector.
+		if w.cfg.SendNotChanged && e.Op == loginp.OpDone {
```

Comment:
I'm uncomfortable using a comparison to an operation that happens to be unused in the switch block and also happens to be the default value, instead of checking the logical conditions along with the other event cases. How about including this in the switch statement as `case prevDesc.Info.Size() == fd.Info.Size() && prevDesc.Info.ModTime() == fd.Info.ModTime()` (optionally removing the second condition into a comment since it's logically redundant if we put it after the other equal-size check)

Reply:
Hum... The semantics I wanted was:
> If all conditions have been checked and no event was detected, then check `w.cfg.SendNotChanged`.

That is why I did not put it in the switch block. I do not see this switch block changing, so I'm not opposed to move it inside the switch block, maybe as the `default` case, so it clearly states: "if nothing else has happened, then check `SendNotChanged`.